samza-core/src/main/java/org/apache/samza/table/remote/couchbase/CouchbaseTableFunctionBase.java

```diff
@@ -38,6 +40,25 @@
   protected String bucketName;
   protected Cluster cluster;
   protected Bucket bucket;
+  protected Class<V> valueClass;
+  protected Serde<V> valueSerde;
+  protected long timeout = 0L;
+  protected TimeUnit timeUnit;
+  protected int ttl = 0; // default value 0 means no ttl, data will be storedforever
+
+  //TODO maybe we can create a builder class to create both read and write functions for the same bucket so that users don't need to type in the same things twice
+  public CouchbaseTableFunctionBase(Class<V> valueClass) {
```

Comment:
There is a tricky way to check the type of a generic type by using anonymous class to avoid java type erasure. We can sync up offline to discuss about the trade-off.

Reply:
http://rgomes-info.blogspot.com/2013/12/using-typetokens-to-retrieve-generic.html

Reply:
Synced offline and didn't to go with the original design of passing in valueClass.